### PR TITLE
Fix/limit order fixes

### PR DESCRIPTION
**Pull request by vinta**
### PR Reminders
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

**Comment by vinta** [comment]
TODO:

- [] @42bchen Suggested to apply https://github.com/ethereum/solidity/issues/9232 to `LIMIT_ORDER_TYPEHASH`

File: contracts/limitOrder/LimitOrderBook.sol [hunk 1]
```diff
@@ -158,7 +179,27 @@ contract LimitOrderBook is
 
         _ordersStatus[orderHash] = ILimitOrderBook.OrderStatus.Cancelled;
 
-        emit LimitOrderCancelled(order.trader, order.baseToken, orderHash);
+        int256 positionSize;
```
Comment (KimiWu123) [inline contracts/limitOrder/LimitOrderBook.sol:182]: question, we need this piece of code bcs we need `positionSize` and `positionNotional` for the event, right? If so, all the info should be able to be found in our backend db, right? Why don't we just fetch these values from DB?

**Review by KimiWu123 — APPROVED** [review]

File: contracts/limitOrder/LimitOrderBook.sol (hunk 1 above)
Comment (vinta) [reply contracts/limitOrder/LimitOrderBook.sol:182]: Yes, limit order backend currently will delete cancelled limit orders though. But I think it's still good to know in subgraph.

Can ask Tempo not to delete cancelled orders, and frontend filters out cancelled ones.
